src/tlo/analysis/life_expectancy.py

```diff
@@ -0,0 +1,305 @@
+        return df.loc[pd.to_datetime(df.date).between(*TARGET_PERIOD)].groupby(["age_group", "sex"])[
+            "person_id"].count()
+
+    return extract_results(
+        results_folder,
+        module="tlo.methods.demography",
+        key="death",
+        custom_generate_series=extract_deaths_by_age_group,
+        do_scaling=False
+    )
+
+
+# get population size in target period
+# get one row of log for each draw/run and compile into df
+def _extract_pop_during_target_period(results_folder: Path, key):
+    info = get_scenario_info(results_folder)
+
+    module = "tlo.methods.demography"
+    key = key
+
+    # Create an empty DataFrame with the specified multi-index columns
+    multi_index_columns = _create_multi_index_columns(results_folder)
+    res = pd.DataFrame(columns=multi_index_columns)
+
+    for draw in range(info['number_of_draws']):
+        for run in range(info['runs_per_draw']):
+            df: pd.DataFrame = load_pickled_dataframes(results_folder, draw, run, module)[module][key]
+            output_from_eval: pd.Series = df.loc[pd.to_datetime(df.date).between(*TARGET_PERIOD)].stack()
+            res.loc[:, (draw, run)] = output_from_eval[1:]
+
+    return res
+
+
+def _get_population_size_by_age(results_folder):
+    def extract_infants_during_target_period(df: pd.DataFrame) -> pd.Series:
+        return df.loc[pd.to_datetime(df.date).between(*TARGET_PERIOD), '0']
+
+    # get children aged=0 in target period
+    num_children = extract_results(
+        results_folder,
+        module="tlo.methods.demography",
+        key="num_children",
+        custom_generate_series=extract_infants_during_target_period,
+        do_scaling=False
+    )
+    num_f0 = num_children / 2  # todo should this be exactly 50/50?
+    num_m0 = num_children / 2
+
+    # get age_range_m and age_range_f for all age-groups
+    num_males = _extract_pop_during_target_period(results_folder=results_folder, key='age_range_m')
+    num_females = _extract_pop_during_target_period(results_folder=results_folder, key='age_range_f')
+    num_males = num_males.reset_index(level=0, drop=True)
+    num_females = num_females.reset_index(level=0, drop=True)
+
+    # then subtract age 0 from age-group 0-4
+    num_males_adj = num_males.iloc[0:1, :] - num_m0.iloc[0:1, :].values
+    # add adjusted data back into dataframe
+    num_males.iloc[0] = num_males_adj.values
+    # change first row index
+    num_males.index.values[0] = '1-4'
+    # add in infant numbers to main dataframe
+    t1 = pd.concat([num_m0, num_males])
+    # set correct index
+    t1.index.values[0] = '0'
+
+    # repeat for females
+    num_females_adj = num_females.iloc[0:1, :] - num_f0.iloc[0:1, :].values
+    num_females.iloc[0] = num_females_adj.values
+    num_females.index.values[0] = '1-4'
+    t2 = pd.concat([num_f0, num_females])
+    t2.index.values[0] = '0'
+
+    # combine pop aged 90+
+    # Sum the last three rows column-wise
+    last_three_rows_sum = t1.iloc[-3:].sum()
```

Comment:
it may be "unsafe" to rely on it being last three rows. the index would have had the ages in (if if wasn't dropped) and then you could do a `Slice` to get everyone above 90.

Reply:
yes good point - removed now